model/model/v1alpha/model.proto

```diff
@@ -55,6 +55,32 @@ message ReadinessResponse {
   common.healthcheck.v1beta.HealthCheckResponse health_check_response = 1;
 }
 
+// State describes the state of a model. See [Deploy
+// Models](https://www.instill.tech/docs/latest/model/deploy) for more
+// information.
+enum State {
+  // Unspecified. This state can occur while a model is under operation.
+  STATE_UNSPECIFIED = 0;
+  // Offline is the state when the model instance number is 0.
+  STATE_OFFLINE = 1;
+  // Scaling is the transition state when a model instance is scaling up or down.
+  STATE_SCALING = 2;
+  // Active is the state when a model is processing requests.
+  STATE_ACTIVE = 3;
+  // Idle is the state when a model is alive but not processing requests.
+  STATE_IDLE = 4;
+  // Error is the state when the model is undeployable.
+  STATE_ERROR = 5;
+}
+
+// Region describes the cloud provider and region that a model is hosted on.
+enum Region {
+  // Unspecified.
+  REGION_UNSPECIFIED = 0;
+  // europe-west-4 on gcp.
+  REGION_GCP_EUROPE_WEST_4 = 1;
```

Comment:
❓ I'm curious why you chose to have an enum over a string for the `region` field. Not that I'm against it, the set of values is limited, but so are the `hardware` values or other fields in the API and we seem to rely on strings quite often.

Reply:
I guess there is no particular reason behind this, the thought process was that we are probably not going to support multi-region for `model` anytime soon, maybe is good to have an enum to allow the user to know the only region we support now.
But I will follow the connector/component design that to have a json definition file to allow frontend to generate auto-form in the future, will make `region` string type now 👍🏼